CI: Conformance Runtime (privileged): TestOps: "0" is not greater than "0"

## CI failure

```
--- FAIL: TestOps (0.01s)
    ops_test.go:57: created dummy0 (index 3)
    ops_test.go:63: qdiscs before: [{LinkIndex: 3, Handle: none, Parent: root, Refcnt: 2}]
    ops_test.go:87: 
        	Error Trace:	/host/pkg/datapath/linux/bandwidth/ops_test.go:87
        	Error:      	"0" is not greater than "0"
        	Test:       	TestOps
```

Link: https://github.com/cilium/cilium/actions/runs/7999342628/job/21847015805

--- a/pkg/datapath/linux/bandwidth/ops_test.go
+++ b/pkg/datapath/linux/bandwidth/ops_test.go
@@ -21,7 +21,6 @@ import (
 
 func freshNetNS(t *testing.T) {
 	runtime.LockOSThread()
-	defer runtime.UnlockOSThread()
 
 	oldNetNS, err := netns.Get()
 	assert.NoError(t, err)
@@ -30,6 +29,7 @@ func freshNetNS(t *testing.T) {
 	t.Cleanup(func() {
 		testNetNS.Close()
 		netns.Set(oldNetNS)
+		runtime.UnlockOSThread()
 	})
 }
 

